src/components/SaveFileButton.js

```diff
@@ -1,74 +1,32 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
-import { remote } from 'electron';
-import fs from 'fs';
+import { saveFile } from 'src/lib/save';
 import Button from './Button';
 
 
 export default class SaveFileButton extends Component {
   static propTypes = {
     contents: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
-    dialogTitle: PropTypes.string,
-    suggestedFileName: PropTypes.string,
-    buttonText: PropTypes.string,
-    onSaved: PropTypes.func
-  }
-
-  constructor(props) {
-    super(props);
-    this.state = {};
+    saveOptions: PropTypes.object,
+    isSaved: PropTypes.bool,
+    onSaved: PropTypes.func,
   }
 
   handleClick() {
-    remote.dialog.showSaveDialog({
-      title: this.props.dialogTitle || 'Save File',
-      defaultPath: this.props.defaultPath,
-    }, this.filePickerCallback.bind(this));
-  }
-
-  filePickerCallback(filename) {
-    if (!filename) {
-      // User didn't pick a file. Noop.
-      return;
-    }
-
-    this.setState({ save: 'saving', to: filename });
-
-    fs.writeFile(
-      filename,
+    saveFile(
       this.props.contents,
-      { mode: '0600' },
-      (error) => {
-        if (error) {
-          return this.setState({ save: 'failed' });
-        }
-
-        this.setState({ save: 'succeeded' });
-        if (this.props.onSaved) {
-          this.props.onSaved(filename);
-        }
-      });
+      this.props.saveOptions,
+      this.props.onSaved
+    );
   }
 
   render() {
-    let text;
-
-    if (this.state.save === 'saving') {
-      text = 'Saving';
-    } else if (this.state.save === 'failed') {
-      text = 'Failed';
-    } else if (this.state.save === 'succeeded') {
-      text = 'Saved';
-    } else {
-      text = this.props.buttonText || 'Save';
-    }
-
```

Comment:
I see there is fairly little reason to delete this logic in the end. You could argue that the saved state is shown as part of the `ShareRow` now, so this changing text is not needed anymore. But you could also argue that the `CopyButton` still has this changing text behavior, so it would be more consistent to keep it...
I'm kind of indifferent, both make sense to me.